allensdk/api/cloud_cache/utils.py

```diff
@@ -0,0 +1,76 @@
+from typing import Optional
+import warnings
+import urllib.parse as url_parse
+import pathlib
+import hashlib
+
+
+def bucket_name_from_uri(uri: str) -> Optional[str]:
+    """
+    Read in a URI and return the name of the AWS S3 bucket it points towards
+
+    Parameters
+    ----------
+    uri: str
+        A generic URI
+
+    Returns
+    -------
+    str
+        An AWS S3 bucket name. Note: if 's3.amazonaws.com' does not occur in
+        the URI, this method will return None and emit a warning.
+    """
+    s3_pattern = '.s3.amazonaws.com'
```

Comment:
what is this requirement?
for example, from `slapp` this was a valid URI:
`s3://prod.slapp.alleninstitute.org/2-line 2-project/20200520051838/1003122317_full_video.webm`

Reply:
These are the URIs I see for objects in my cartoon prototyping bucket
```
https://danielsf-vba-versioning-prototype.s3.amazonaws.com/txt_file.txt?versionId="7nv3u.Y6PA3vjBsSVeL3fGHzLdvzdYM5"
```

I believe we are trying to use the URI/URLs that allow a user to access resources with generic HTTP GET requests. I don't think you can use `s3://` URLs that way?

Should I specify that the URI/URL we expect is "suitable for a generic GET request?"

Reply:
oh I see, yes, i forgot this was via http.
for a similar one the link is:
https://s3-us-west-2.amazonaws.com/prod.slapp.alleninstitute.org/2-line+2-project/20200520051838/1002314807_full_video.webm
it still doesn't meet your pattern, because of the `us-west-2`
I'm not sure when that location shows up and when it doesn't.

Reply:
Not sure this makes anything clearer:
```
https://s3-us-west-2.amazonaws.com/danielsf-vba-versioning-prototype/txt_file.txt
https://danielsf-vba-versioning-prototype.s3.amazonaws.com/txt_file.txt
```
are both valid ways to download the same resource from S3

Reply:
knowing is half the battle

Reply:
According to this
https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/
they are deprecating `https://s3-[region]-amazonaws.com/....` URLs.

There may still be some ambiguity between
```
https://bucket-name.s3.amazonaws.com/object_name
```
and
```
https://bucket-name.s3-region-name.amazonaws.com/object_name
```

I'll see if I can adapt the method to recognize both (shouldn't be hard)

Reply:
`bucket_name_from_uri` (soon to be renamed `bucket_name_from_url`) now uses a regex to detect the bucket name in either valid URL formulation.